src/main/java/org/zalando/nakadi/controller/EventPublishingController.java

```diff
@@ -82,9 +85,21 @@ private ResponseEntity postEventInternal(final String eventTypeName,
             final JSONArray eventsAsJsonObjects = new JSONArray(eventsAsString);
 
             final int eventCount = eventsAsJsonObjects.length();
-            eventTypeMetrics.reportSizing(eventCount, eventsAsString.length());
+            final EventPublishResult result = publisher.publish(eventsAsJsonObjects, eventTypeName, client);
+            if (result.getStatus() == EventPublishingStatus.SUBMITTED) {
+                eventTypeMetrics.reportSizing(eventCount, eventsAsString.getBytes(StandardCharsets.UTF_8).length);
+            } else if (result.getStatus() == EventPublishingStatus.FAILED && eventCount != 0) {
+                final int successfulBatches = result.getResponses()
+                        .stream()
+                        .filter(r -> r.getPublishingStatus() == EventPublishingStatus.SUBMITTED)
+                        .collect(Collectors.toList())
+                        .size();
+                eventTypeMetrics.reportSizing(successfulBatches,
+                        (eventsAsString.getBytes(StandardCharsets.UTF_8).length / eventCount) * successfulBatches);
```

Comment:
I'm not sure about the precision here.
AFAIR if you divide int by int you will get int.